app/src/androidTest/java/org/akvo/flow/activity/form/emptyformsview/FormActivityTest.java

```diff
@@ -0,0 +1,500 @@
+import static org.akvo.flow.activity.form.FormActivityTestUtil.getDateButton;
+import static org.akvo.flow.activity.form.FormActivityTestUtil.getDateEditText;
+import static org.akvo.flow.activity.form.FormActivityTestUtil.getDoubleEntryInput;
+import static org.akvo.flow.activity.form.FormActivityTestUtil.getFormActivityIntent;
+import static org.akvo.flow.activity.form.FormActivityTestUtil.getFreeTextInput;
+import static org.akvo.flow.activity.form.FormActivityTestUtil.getGeoButton;
+import static org.akvo.flow.activity.form.FormActivityTestUtil.getMediaButton;
+import static org.akvo.flow.activity.form.FormActivityTestUtil.getOptionView;
+import static org.akvo.flow.activity.form.FormActivityTestUtil.getQuestionHeader;
+import static org.akvo.flow.activity.form.FormActivityTestUtil.selectAndVerifyTab;
+import static org.akvo.flow.activity.form.FormActivityTestUtil.verifyAccuracyLabel;
+import static org.akvo.flow.activity.form.FormActivityTestUtil.verifyDoubleEntryTitle;
+import static org.akvo.flow.activity.form.FormActivityTestUtil.verifyGeoLabel;
+import static org.akvo.flow.activity.form.FormActivityTestUtil.verifyHelpTip;
+import static org.akvo.flow.activity.form.FormActivityTestUtil.verifyQuestionHeader;
+import static org.akvo.flow.activity.form.FormActivityTestUtil.verifyToolBar;
+import static org.akvo.flow.activity.form.FormActivityTestUtil.withQuestionViewParent;
+import static org.akvo.flow.tests.R.raw.all_questions_form;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.core.IsNot.not;
+
+@LargeTest
+@RunWith(AndroidJUnit4.class)
+public class FormActivityTest {
+
+    private static final String FORM_TITLE = "Test form";
+    private static SurveyInstaller installer;
+    private static Survey survey;
+
+    @Rule
+    public ActivityTestRule<FormActivity> rule = new ActivityTestRule<FormActivity>(
+            FormActivity.class) {
+        @Override
+        protected Intent getActivityIntent() {
+            return getFormActivityIntent(155852013L, "156792013", FORM_TITLE, 0L, false);
+        }
+    };
+
+    @BeforeClass
+    public static void beforeClass() {
+        Context targetContext = InstrumentationRegistry.getTargetContext();
+        SurveyRequisite.setRequisites(targetContext);
+        installer = new SurveyInstaller(targetContext);
+        survey = installer.installSurvey(all_questions_form, InstrumentationRegistry.getContext());
+    }
+
+    @After
+    public void afterEachTest() {
+        installer.deleteResponses(TEST_FORM_SURVEY_INSTANCE_ID);
+    }
+
+    @AfterClass
+    public static void afterClass() {
+        SurveyRequisite.resetRequisites(InstrumentationRegistry.getTargetContext());
+        installer.clearSurveys();
+    }
+
+    @Test
+    public void viewEmptySurveyTest() {
+        //make sure everything is loaded
+        addExecutionDelay(5000);
```

Comment:
I was trying to find out where there is an example to use `IdlingResources` in order to avoid having to sleep.

Reply:
From what I understand you need to implement them in app code which I do  not like. I will try removing the execution delay and see if it works.